src/pages/layout/ConnectWallet.tsx

```diff
@@ -0,0 +1,125 @@
+import React, {useState} from "react";
+import {useWallet} from "@aptos-labs/wallet-adapter-react";
+import {
+  Button,
+  Collapse,
+  Dialog,
+  List,
+  ListItemButton,
+  ListItemText,
+  Stack,
+  Tooltip,
+  Typography,
+} from "@mui/material";
+import ExpandLess from "@mui/icons-material/ExpandLess";
+import ExpandMore from "@mui/icons-material/ExpandMore";
+import {truncateAddress} from "../utils";
+
+const COPIED_TOOLTIP = 2000; // 2s
+
+export default function Wallet() {
+  const {wallets, connect, connected, account, disconnect} = useWallet();
+  const [walletConnectOpen, setWalletConnectOpen] = useState<boolean>(false);
+  const [walletDropdownOpen, setWalletDropdownOpen] = useState<boolean>(false);
+  const [copiedTooltipOpen, setCopiedTooltipOpen] = useState<boolean>(false);
+  const handleConnectWalletClick = () => {
+    setWalletConnectOpen(true);
+  };
+  const handleWalletDropdownClick = () => {
+    setWalletDropdownOpen(!walletDropdownOpen);
+  };
+  const copyAddress = async () => {
+    await navigator.clipboard.writeText(account?.address!);
+
+    setCopiedTooltipOpen(true);
+
+    setTimeout(() => {
+      setCopiedTooltipOpen(false);
+    }, COPIED_TOOLTIP);
+  };
+
+  return (
+    <>
+      {connected ? (
```

Comment:
Wallet adapter has a UI package that can help with all of that

https://github.com/aptos-labs/aptos-wallet-adapter/tree/main/packages/wallet-adapter-ant-design